### ISSUE TITLE
Allow Actions and Responders to be injected

### ISSUE THREAD

**Issue by mvriel** [somos/framework#27]
Currently the Action class accepts a callable to execute business logic related to an action that you want to do in your application; but often you need some dependencies to execute your Action.

In order to do this it would be awesome if we can construct a callable object (one that has __invoke() as a method) and have PHP-DI automatically inject any dependencies in the constructor. 

A possible way to have the user do this is using the following syntax:

```
$somos = Somos::start()
    ->add(Action::get(Route::create('/login')->post(), SignIn::class)->respondWith(Redirect::class, ['url' => '/login']))
    ->handle(new HandleRequest());
```

You can see the 'SignIn::class' and 'Redirect::class' statements, it seems logical to use a ActionFactory and ResponderFactory behind the scenes that can retrieve the services from the container

> Note: the syntax above may change over the course of the new few weeks; I am still experimenting with what would be a good syntax and feels intuitive

**Comment by harikt** [comment]
Hi @mvriel , 

I wonder whether you like the work on https://github.com/friendsofaura/FOA.Responder_Bundle and may be changed to make use of psr-7 response will help ?

**Comment by mvriel** [comment]
Hi @harikt, so far I have made a few incompatible design decisions but I will keep that library in mind as I like how it brings multiple templating engines together.

**Comment by mvriel** [comment]
The requirements described in this issue have been resolved in #31, this issue is as such closed

**Comment by harikt** [comment]
Sure @mvriel :) .
